src/helpers.ts

```diff
@@ -61,24 +61,31 @@ function hasNodeExecutable(rootPath: string, executable: string): boolean {
  * @returns {string}
  */
 export function pathToJest(pluginSettings: IPluginSettings) {
+  if (isDefaultSettingForPathToJest(pluginSettings.pathToJest) && isBootstrappedWithCRA(pluginSettings.rootPath)) {
+    return 'npm test --'
+  }
+
   if (pluginSettings.pathToJest) {
-    if (isBootstrappedWithCRA(pluginSettings.rootPath)) {
-      return 'npm test --'
-    }
     return normalize(pluginSettings.pathToJest)
   }
 
-  const platform = process.platform
-  if (platform === 'win32' && existsSync(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest.cmd'))) {
-    return normalize(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest.cmd'))
-  } else if (
-    (platform === 'linux' || platform === 'darwin') &&
-    existsSync(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest'))
-  ) {
-    return normalize(join(pluginSettings.rootPath, 'node_modules', '.bin', 'jest'))
+  const localJestExecutable = pathToLocalJestExecutable(pluginSettings.rootPath)
+  if (existsSync(localJestExecutable)) {
+    return localJestExecutable
   }
+  return `jest${isWindows() ? '.cmd' : ''}`
+}
+
+function isDefaultSettingForPathToJest(str) {
+  return str === null
```

Comment:
It practically doesn't make any large difference, but wouldn't it be nicer if we change the default value of `jest.pathToJest` within package.json to `""` instead of `null`, like `jest.pathToConfig` and `jest.rootPath`?

Reply:
I'm going to call you out on a pattern in your comments @stephtr. I find it frustrating when you structure your feedback as "wouldn't it be better if ..." or "wouldn't it be nicer if ..." and don't explain why you favour your suggestion. If you provided an explanation I'd be willing to discuss the options. Without one it comes across to me like you know better, and I should too.

For these default values feel free to change them to whatever you want in another PR. It doesn't make any difference to me. I'd lean towards not changing the default JavaScript behaviour and have values default to `undefined` when they're not provided, or maybe have a look if VS Code suggests any other conventions with their configuration API.